File: BAEKJOON/Python/5_1_dimension_array/Prob_8958.py
N = int(input())
Array = []
tempArray = []
Point = 0
Score = 0
for i in range(0,N,1):
    Array = input()
    length = len(Array)
    for i in range(0,length,1):
        if (Array[i] == 'O'):
            if (Array[i-1]==Array[i]):
                Point = Point + 1
            else:
                Point = 1
        else:
            Point = 0
        Score = Point + Score
    print(Score)
    Score = 0
    Point = 0
    Array = []

File: BAEKJOON/C/5_1_dimension_array/Prob_8958.cpp
#include <stdio.h>
#include <string.h>

int main()
{
	int N = 0u;
	int Point = 1u;
	char Array[80] = {0, };
    char preArray = 0;
	int length = 0u;
	int sum = 0u;

	scanf("%d", &N);

	for (int i = 0; i < N; i++){
		scanf("%s", &Array);	// scanf_s는 메모리 크기도 input으로 넣어줘야댐
		length = sizeof(Array) / sizeof(char);

		for (int i = 0; i < length; i++){
			if ((Array[i] == 'O')){
				if (Point == 0){
					Point = 1u;
				}
				if (Array[i] == preArray){
					Point = Point + 1u;
				}

			}
			if ((Array[i] == 'X') || (Array[i] == 0)){
				Point = 0u;
			}
			sum = Point + sum;
			preArray = Array[i];
		}
		printf("%d\n", sum);
		sum = 0u;
		Point = 1u;
		preArray = 0;
	}

}